internal-dns-cli/src/bin/dnswait.rs

```diff
@@ -0,0 +1,98 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
Just to make sure I'm clear - the main reason for using this instead of `dig +short <name>` is the timeout?

Reply:
Yeah, I had `dig` at first, but I realized it will only try a handful of times and only wait a few seconds, _and_ you can't tell from its exit status that it's failed.  I needed something that's going to retry an arbitrary number of times, waits until it finds some results, too, and if it's going to fail, fails clearly.

Reply:
Not for this PR, but maybe it makes sense to move this utility into the omicron-brand zone alongside some of the other network setup that might be more general? Seems a bit like a pain-in-the-butt to manually package it into each zone that could need it